app/render/audioplaybackcache.cpp

```diff
@@ -64,10 +64,14 @@ void AudioPlaybackCache::WritePCM(const TimeRange &range, SampleBufferPtr sample
 
   // Ensure if we have enough segments to write this data, creating more if not
   qint64 length_diff = params_.time_to_bytes(range.out()) - playlist_.GetLength();
-  while (length_diff > 0) {
-    qint64 seg_sz = qMin(kDefaultSegmentSize, length_diff);
-    playlist_.push_back(CreateSegment(seg_sz, playlist_.GetLength()));
-    length_diff -= seg_sz;
+  qint64 whole_segments = length_diff / kDefaultSegmentSize;
+  qint64 remainder = length_diff % kDefaultSegmentSize;
+
+  for (int i = 0; i < whole_segments; i++) {
+    playlist_.push_back(CreateSegment(kDefaultSegmentSize, playlist_.GetLength()));
+  }
+  if (remainder > 0) {
+    playlist_.push_back(CreateSegment(remainder, playlist_.GetLength()));
```

Comment:
Does this actually make a difference? It looks like it does the exact same thing but with more lines of code.

Reply:
It did seem too, I'll try and redo the tests and post them here. As I understand it, this removes the repeated qmin and probably allows a bit of branch prediction with the if statement. Happy to be corrected though :)

Reply:
Hmm perhaps. Yes I'll be interested to see the test information.

Reply:
I can't actually get a repeat test so I suspect it was a fluke, shall I revert the commit?